src/app/components/students/adding-student-page/adding-student/adding-student.component.ts

```diff
@@ -0,0 +1,218 @@
+      this.lastNameInfo = this.messageForLastName[0];
+      this.lastNameCorrectness = false;
+    } else if (this.spaceRegExp.test(valueItem) && valueItem) {
+      this.lastNameInfo = this.messageForLastName[1];
+      this.lastNameCorrectness = false;
+    } else if (!this.firstLetterUppercaseRegExp.test(valueItem) && valueItem) {
+      this.lastNameInfo = this.messageForLastName[2];
+      this.lastNameCorrectness = false;
+    } else if (!this.spaceRegExp.test(valueItem) &&
+      ((this.lastNameRegExp.test(valueItem)
+        && this.firstLetterUppercaseRegExp.test(valueItem)) || !valueItem)) {
+      this.lastNameInfo = "";
+      this.lastNameCorrectness = true;
+    }
+  }
+
+  checkAgeCorrectness(valueItem: any): void {
+    if (!this.ageRegExp.test(valueItem) && valueItem) {
+      this.ageInfo = this.messageForAge[0];
+      this.ageCorrectness = false;
+    } else if (this.spaceRegExp.test(valueItem) && valueItem) {
+      this.ageInfo = this.messageForAge[1];
+      this.ageCorrectness = false;
+    } else if (!this.spaceRegExp.test(valueItem) &&
+      (this.ageRegExp.test(valueItem) || !valueItem)) {
+      this.ageInfo = "";
+      this.ageCorrectness = true;
+    }
+  }
+
+  checkAddressCorrectness(valueItem: any): void {
+    if (!this.addressRegExp.test(valueItem) && valueItem) {
+      this.addressInfo = this.messageForAddress[0];
+      this.addressCorrectness = false;
+    } else if (!this.firstLetterUppercaseRegExp.test(valueItem) && valueItem) {
+      this.addressInfo = this.messageForAddress[1];
+      this.addressCorrectness = false;
+    } else if (this.addressRegExp.test(valueItem)
+      && this.firstLetterUppercaseRegExp.test(valueItem)) {
+      this.addressInfo = "";
+      this.addressCorrectness = true;
+    }
+  }
+
+  checkNameLengthCondition(name: string): boolean {
+    if (!name) {
+      this.nameInfo = this.messageForName[5];
+      return false;
+    } else if (name.length < 2) {
+      this.nameInfo = this.messageForName[3];
+      return false;
+    } else if (name.length > 15) {
+      this.nameInfo = this.messageForName[4];
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  checkLastNameLengthCondition(lastName: string): boolean {
+    if (!lastName) {
+      this.lastNameInfo = this.messageForLastName[5];
+      return false;
+    } else if (lastName.length < 2) {
+      this.lastNameInfo = this.messageForLastName[3];
+      return false;
+    } else if (lastName.length > 20) {
+      this.lastNameInfo = this.messageForLastName[4];
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  checkAgeRestrictionsCondition(age: number | null): boolean {
+    if (!age) {
+      this.ageInfo = this.messageForAge[4];
+      return false;
+    } else if (age < 17) {
+      this.ageInfo = this.messageForAge[2];
+      return false;
+    } else if (age > 24) {
+      this.ageInfo = this.messageForAge[3];
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
I think there is redundant 'else'. 

Will such a solution be sufficient in our case:
if(!age) {}
if(age < 17) {}
.... and so on?
What do you think?